internal/collectors/cpu.go

```diff
@@ -58,3 +63,56 @@ func parseCPUSocket(content string) (int, int) {
 	*/
 	return cpu + 1, socket + 1
 }
+
+// Add architecture-specific fields into the given result. Note that this could
+// have also been implemented by adding specific `gobuild` tags into
+// architecture-specific files, but this would needlessly complicate testing.
+func addArchExtras(result Result) Result {
+	if runtime.GOARCH == "arm64" {
```

Comment:
Why not rely on the given `arch` argument of `run()` for it. Then it would be more consistent. Otherwise you can run the collectors with an arbitrary arch but this check actually checks the compiled ABI.